File: 2025/c#/day-1/part-1.cs
public static class Part1
{
  public static int Solve(string input)
  {
    var lines = input.Split('\n');

    var dialValue = 50;
    var zeroCount = 0;

    var values = lines.Select(x => int.Parse(x.Substring(1)) * (x.StartsWith("L") ? -1 : 1));

    foreach (var value in values)
    {
      dialValue += value;
      if (dialValue % 100 == 0) zeroCount++;
    }

    return zeroCount;
  }
}

File: 2025/c/day-1/part-1.c
#include <stdio.h>
#include <string.h>

int main()
{
  FILE *fptr = fopen("input.txt", "r");

  char lr;
  int value;

  int zeroCount = 0;
  int dialValue = 50;

  while (fscanf(fptr, " %c%d", &lr, &value) == 2) {
    dialValue += value * (lr == 'L' ? -1 : 1);
    if (dialValue % 100 == 0) zeroCount++;
  }

  fclose(fptr);

  printf("ZeroCount: %d", zeroCount);

  return 0;
}

